man/fi_pingpong.1.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: page
+title: fi_pingpong(1)
+tagline: Libfabric Programmer's Manual
+---
+{% include JB/setup %}
+
+
+# NAME
+
+fu_pingpong  \- Quick and simple pingpong test for libfabric
+
+
+# SYNOPSYS
+
+```
+ fi_pingpong [OPTIONS] -s <server IP>	start server
+ fi_pingpong [OPTIONS] <server IP>	connect to server
+```
+
+
+# DESCRIPTION
+
+`fi_pingpong` is a generic pingpong test allowing provider authors and libfabric users to test
+the core feature of the libfabric library: transmitting data between two nodes.
+Moreover it agregates some data during the test on each end in order to diplay some statistics
+after each test has been run, and potentially ask to verify data consistency.
+
+An endpoint can be specified to be used as the data endpoint. By default, the DGRAM endpoint
+is always selected. Therefore it is designed to transmit data from one node toward the other
+over the selected endpoint, or the default DGRAM if none is selected.
+
+
+# HOW TO RUN TESTS
+
+To launch properly a pingpong test, two termimals have to be open : one for the client,
+the other for the server. However the server must be already running to start the client.
+
+As a client-server test, each have the following usage model:
+
+server$	fi_pingpong -s <server endpoint address>	: start server
+client$	fi_pingpong <server endpoint address>		: connect to server
+
+
+# OPTIONS
+
+The command line of the server and the client must be launched with exactly the same set of options. Otherwise the test would fail because the two nodes would not be able to communicate properly.
```

Comment:
For the moment, I'm not worried about making this program bullet proof.  I'd really much rather get this posted to the community for review.

All other apps require the same CLI options for client/server -- so let's be ok with that for now.